src/main/java/com/google/devtools/build/lib/skyframe/ToolchainUtil.java

```diff
@@ -108,6 +116,14 @@ static ToolchainContext createToolchainContext(
             .addAll(registeredExecutionPlatforms.registeredExecutionPlatformKeys())
             .add(hostPlatformKey)
             .build();
+
+    // Filter out execution platforms that don't satisfy the extra constraints.
+    availableExecutionPlatformKeys =
```

Comment:
I would expect this to use a streamable approach with a filter function. I guess that is tricky because of the SkyFunction pattern of returning "null" everywhere to signal incomplete inputs. I wish we used an exception type instead to be able to write better code...